### PR TITLE
✨ Destination BigQuery - Handle Schema Changes

### LINKED ISSUE #27590: BigQuery and Snowflake do not allow alter table statements which change a columns data type
we should handle schema changes similarly to normalization. figure out how to do that.

### PR DESCRIPTION
**Pull request by jbfbell**
# Handling Schema chages
We want to detect whether the schema which is generated by the stream configuration matches the existing table's schema.
In the short term we've decided that _any_ mismatch in the schema triggers a soft reset, or a rebuild of the final table from the raw table. 

This only matters for non-overwrite syncs.

### CODE REVIEW

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java [hunk 1]
```diff
@@ -179,7 +188,7 @@ public String createTable(final StreamConfig stream, final String suffix) {
             """
             CREATE SCHEMA IF NOT EXISTS ${final_namespace};
 
-            CREATE TABLE ${final_table_id} (
+            CREATE OR REPLACE TABLE ${final_table_id} (
```
Comment (jbfbell) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java:191]: @edgao calling out that this wasn't exactly part of my mission with this but if a sync fails or is cancelled and the tmp table is never cleaned up, without the `REPLACE` statement the sync will fail. Let me know if this is too sneaky/dangerous

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`eca9abe28f`](https://github.com/airbytehq/airbyte/pull/28382/commits/eca9abe28f4767532c735899887410b6c10e869e)) - ✅

⏲️  Total pipeline duration: 29mn59s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/josephbell_27590_schema-changes/1689698985/eca9abe28f4767532c735899887410b6c10e869e/destination-bigquery/1.5.4/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`eca9abe28f`](https://github.com/airbytehq/airbyte/pull/28382/commits/eca9abe28f4767532c735899887410b6c10e869e)) - ✅

⏲️  Total pipeline duration: 17mn50s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/josephbell_27590_schema-changes/1689698985/eca9abe28f4767532c735899887410b6c10e869e/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`eca9abe28f`](https://github.com/airbytehq/airbyte/pull/28382/commits/eca9abe28f4767532c735899887410b6c10e869e)) - ✅

⏲️  Total pipeline duration: 22mn13s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/josephbell_27590_schema-changes/1689702000/eca9abe28f4767532c735899887410b6c10e869e/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

File: airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/SqlGenerator.java [hunk 2]
```diff
@@ -4,10 +4,16 @@
...
 
+
+import java.util.List;
 import java.util.Optional;
+import java.util.Set;
 
 public interface SqlGenerator<DialectTableDefinition> {
 
+  Set<String> FINAL_TABLE_AIRBYTE_COLUMNS = Set.of("_airbyte_raw_id", "_airbyte_extracted_at", "_airbyte_meta");
+  String SOFT_RESET_SUFFIX = "_ab_soft_reset";
```
Comment (evantahler) [inline airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/SqlGenerator.java:15]: nit: Maybe these should be static constants somewhere

Comment (jbfbell) [reply airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/SqlGenerator.java:15]: IDE yelled at me: all interface members are "public, static, final" by default so static keyword is redundant

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java [hunk 3]
```diff
@@ -190,32 +199,120 @@ PARTITION BY (DATE_TRUNC(_airbyte_extracted_at, DAY))
...
+        clusterColumns.add(columnId.name());
+      });
+    }
+    clusterColumns.add("_airbyte_extracted_at");
+    return clusterColumns;
+  }
 
-      }
-    } else {
-      // TODO idk
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java:204]: the end of an era

File: airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/SqlGenerator.java [hunk 4]
```diff
@@ -60,4 +73,5 @@ public interface SqlGenerator<DialectTableDefinition> {
    */
   Optional<String> overwriteFinalTable(String finalSuffix, StreamConfig stream);
 
+  AlterTableReport buildAlterTableReport(final StreamConfig stream, final DialectTableDefinition existingTable);
```
[the remark this answers is not in the document]
Comment (edgao) [reply airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/SqlGenerator.java:76]: nit: also, is this only used inside the sqlgenerator itself? I'd prefer to keep it out of the interface if so, even if we still want to provide the AlterTableReport record for convenience

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java [hunk 5]
```diff
@@ -190,32 +199,120 @@ PARTITION BY (DATE_TRUNC(_airbyte_extracted_at, DAY))
...
+
+    return new AlterTableReport(columnsToAdd, columnsToRemove, columnsToChangeType, isDestinationV2Format);
+  }
+
+  @Override
+  public List<String> softReset(final StreamConfig stream) {
+    String createTempTable = createTable(stream, SOFT_RESET_SUFFIX);
+    String clearLoadedAt = clearLoadedAt(stream.id());
+    String rebuildInTempTable = updateTable(SOFT_RESET_SUFFIX, stream);
+    String overwriteFinalTable = overwriteFinalTableStatement(stream, SOFT_RESET_SUFFIX);
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java:307]: nice, I didn't even realize we could reuse these methods like this

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java [hunk 6]
```diff
@@ -190,32 +199,120 @@ PARTITION BY (DATE_TRUNC(_airbyte_extracted_at, DAY))
...
+    final Set<String> columnsToRemove = existingSchema.keySet().stream()
+            .filter(name -> !containsIgnoreCase(streamSchema.keySet(), name) && !containsIgnoreCase(FINAL_TABLE_AIRBYTE_COLUMNS, name))
+            .collect(Collectors.toSet());
+
+    // Columns that are typed differently than the StreamConfig
+    final Set<String> columnsToChangeType = streamSchema.keySet().stream()
+            // If it's not in the existing schema, it should already be in the columnsToAdd Set
+            .filter(name -> {
+              // Big Query Columns are case-insensitive
+              final String matchedKey = matchingKey(existingSchema.keySet(), name).orElse(name);
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java:291]: I found this lambda kind of hard to follow, maybe better to be explicit:
```java
Optional<String> matchedKey = matchingKey(existingSchema.keySet(), name);
if (!matchedKey.isPresent()) { return false; }
return !existingSchema.get(matchedKey.get()).equals(streamSchema.get(name));
```

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java [hunk 7]
```diff
@@ -190,32 +199,120 @@ PARTITION BY (DATE_TRUNC(_airbyte_extracted_at, DAY))
...
+              return !existingSchema.getOrDefault(matchedKey, streamSchema.get(name)).equals(streamSchema.get(name));
+            })
+            .collect(Collectors.toSet());
+
+    final boolean isDestinationV2Format = FINAL_TABLE_AIRBYTE_COLUMNS.stream()
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java:296]: do we have a testcase that exercises this?

**Review by edgao — COMMENTED** [review]
this is awesome! I think it's worth adding something to BaseTypingDedupingTest for this (i.e. run a sync, modify the schema, run a second sync, assert the new records)

had a couple minor comments around code structure. I agree with the refactors you've done here, but we should get this merged in reasonably quickly because there are some conflicts with what I'm doing (mostly because I want to move some of your code into a class that I haven't written yet)

beyond that, just a couple readability comments.

File: airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/SqlGenerator.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (jbfbell) [reply airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/SqlGenerator.java:76]: fair point, removing

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQueryDestinationHandler.java [hunk 8]
```diff
@@ -93,4 +97,38 @@ public void execute(final String sql) throws InterruptedException {
...
+   */
+  public void prepareFinalTable(final BigQuerySqlGenerator sqlGenerator, final StreamConfig stream, final TableDefinition existingTable) {
+    try {
+      if (!sqlGenerator.existingSchemaMatchesStreamConfig(stream, existingTable)) {
+        attemptSoftReset(sqlGenerator, stream);
+      } else {
+        LOGGER.info("Existing Schema matches expected schema, no alterations needed");
+      }
+    } catch (TableNotMigratedException nm) {
+      throw new RuntimeException("Cannot complete destinations v2 sync, final table not migrated: %s".formatted(stream.id().finalName()));
```
[the remark this answers is not in the document]
Comment (jbfbell) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQueryDestinationHandler.java:114]: It didn't extend runtime exception so I had to handle it, updated to extend runtime exception.

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`39dae3f334`](https://github.com/airbytehq/airbyte/pull/28382/commits/39dae3f334b309e5a566d04a36abe2cdc16bd9f9)) - ✅

⏲️  Total pipeline duration: 29mn44s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/josephbell_27590_schema-changes/1689795225/39dae3f334b309e5a566d04a36abe2cdc16bd9f9/destination-bigquery/1.5.4/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java (hunk 6 above)
Comment (jbfbell) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java:291]: How does this look
```java
    // Columns that are typed differently than the StreamConfig
    final Set<String> columnsToChangeType = streamSchema.keySet().stream()
            // If it's not in the existing schema, it should already be in the columnsToAdd Set
            .filter(name -> {
              // Big Query Columns are case-insensitive, first find the correctly cased key if it exists
              return matchingKey(existingSchema.keySet(), name)
                      // if it does exist, only include it in this set if the type (the value in each respective map)
                      // is different between the stream and existing schemas
                      .map(key -> !existingSchema.getOrDefault(key, streamSchema.get(name)).equals(streamSchema.get(name)))
                      // if there is no matching key, then don't include it because it is probably already in columnsToAdd
                      .orElse(false);
            })
            .collect(Collectors.toSet());
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`39dae3f334`](https://github.com/airbytehq/airbyte/pull/28382/commits/39dae3f334b309e5a566d04a36abe2cdc16bd9f9)) - ❌

⏲️  Total pipeline duration: 18mn15s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/josephbell_27590_schema-changes/1689795225/39dae3f334b309e5a566d04a36abe2cdc16bd9f9/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java (hunk 6 above)
Comment (edgao) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java:291]: is the getOrDefault still needed? since we're only entering that map lambda if we found column in existingSchema

but yeah this makes a lot more sense!

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`d2590244bb`](https://github.com/airbytehq/airbyte/pull/28382/commits/d2590244bbd691464b40219baf86693054d60d73)) - ✅

⏲️  Total pipeline duration: 23mn32s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/josephbell_27590_schema-changes/1689807941/d2590244bbd691464b40219baf86693054d60d73/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`d2590244bb`](https://github.com/airbytehq/airbyte/pull/28382/commits/d2590244bbd691464b40219baf86693054d60d73)) - ✅

⏲️  Total pipeline duration: 29mn57s 

| Step   | Result   |
|--------|----------|

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/josephbell_27590_schema-changes/1689807941/d2590244bbd691464b40219baf86693054d60d73/destination-bigquery/1.5.4/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java (hunk 6 above)
Comment (jbfbell) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java:291]: oh yeah good call!

File: airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/SqlGenerator.java [hunk 9]
```diff
@@ -16,21 +22,28 @@ public interface SqlGenerator<DialectTableDefinition> {
...
+   */
+  boolean existingSchemaMatchesStreamConfig(final StreamConfig stream, final DialectTableDefinition existingTable) throws TableNotMigratedException;
+
+  /**
+   * SQL Statements which will rebuild the final table using the raw table data
+   * @param stream the stream to rebuild
+   * @return an ordered sequence of SQL statements to execute to rebuild the final table.
    */
-  String alterTable(final StreamConfig stream, DialectTableDefinition existingTable);
+  List<String> softReset(final StreamConfig stream);
```
[the remark this answers is not in the document]
Comment (jbfbell) [reply airbyte-integrations/bases/base-typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/SqlGenerator.java:49]: FWIW I think we probably _should_ make this optimization but I'm going to hold off for now since the naive implementation of concating with newlines and `;` is failing 😢

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java (hunk 7 above)
Comment (jbfbell) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQuerySqlGenerator.java]: Added a unit test (and moved it to a different class), but should probably add an integration test

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`650405d5e4`](https://github.com/airbytehq/airbyte/pull/28382/commits/650405d5e49f9c33dab94a11e5513cebe7f9b090)) - ❌

⏲️  Total pipeline duration: 24mn04s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/josephbell_27590_schema-changes/1689811267/650405d5e49f9c33dab94a11e5513cebe7f9b090/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

**Review by edgao — APPROVED** [review]
:shipit:

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`650405d5e4`](https://github.com/airbytehq/airbyte/pull/28382/commits/650405d5e49f9c33dab94a11e5513cebe7f9b090)) - ✅

⏲️  Total pipeline duration: 36mn34s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/josephbell_27590_schema-changes/1689811267/650405d5e49f9c33dab94a11e5513cebe7f9b090/destination-bigquery/1.5.4/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`077a5b84d7`](https://github.com/airbytehq/airbyte/pull/28382/commits/077a5b84d75423b092ff93cce7c7fe03f9e355c4)) - ❌

⏲️  Total pipeline duration: 18mn03s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/josephbell_27590_schema-changes/1689815004/077a5b84d75423b092ff93cce7c7fe03f9e355c4/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`4de85d544b`](https://github.com/airbytehq/airbyte/pull/28382/commits/4de85d544bb6a30806df8ad2f138ebed3e2d4f19)) - ✅

⏲️  Total pipeline duration: 27mn37s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/josephbell_27590_schema-changes/1689863846/4de85d544bb6a30806df8ad2f138ebed3e2d4f19/destination-bigquery/1.5.4/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`4de85d544b`](https://github.com/airbytehq/airbyte/pull/28382/commits/4de85d544bb6a30806df8ad2f138ebed3e2d4f19)) - ✅

⏲️  Total pipeline duration: 20mn02s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/josephbell_27590_schema-changes/1689863846/4de85d544bb6a30806df8ad2f138ebed3e2d4f19/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```
